### DISCUSSION TITLE
.env trouble

### DISCUSSION THREAD

**Discussion by Reus1196** [smartcontractkit/full-blockchain-solidity-course-py#1397] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/98214304/163018342-c7d588ce-0d55-4f6c-b546-2cedf37cba55.png)
I was working OK with my .env and then it just started showing errors that I cannot solve. Any help?

**Comment by cromewar** [comment]
Hello there, did you install any extension recently related to code formatting/highlighting?

**Reply by Reus1196** [reply]
Nop, it was all previously installed.

**Reply by cromewar** [reply]
Really strange error, could you please share a screenshot of the folder structure on your project, and also.. does this happens with a nother instance of python?

**Reply by Reus1196** [reply]
![image](https://user-images.githubusercontent.com/98214304/163215704-b35b2f3a-2094-44b7-a159-4148f901defe.png)
![image](https://user-images.githubusercontent.com/98214304/163215751-0d8efd3f-df86-42ed-8b26-e8eed0ef910d.png)
![image](https://user-images.githubusercontent.com/98214304/163215791-630fede1-ccb4-4e4d-b010-34fee8429e49.png)

I am using code as on the video, it is the first time I use .env files and it looks to me as if something is going wrong with it

**Reply by cromewar** [reply]
Just one question does the code works anyway? or it stops you from execution?

**Reply by Reus1196** [reply]
Yes it works but it deploys on the brownie development network as it does not detect the .env, neither the acces to the Rinkeby or other nets

**Reply by cromewar** [reply]
This is really weird as I'm almost sure something is messing up with your vscode installation, did you considerar try this with WSL? there is an article on the course index about how to setup it, just look for "windows support".

**Reply by Reus1196** [reply]
Quite strange cause I did a restart from fabric to PC, installed again all but problem persists, but I just deployed using brownie run scripts/deploy.py --network rinkeby and it worked well and .env info was correctly called and used, but files still showing the same problem... but it works

**Reply by cromewar** [reply]
Well even vscode is complaining the code compiles so is a half win I guess.

**Comment by Noveleader** [comment]
Hey did you install python [dotenv](https://pypi.org/project/dotenv/)?

**Reply by Reus1196** [reply]
Yes v0.16.0

**Comment by harshuo2** [comment]
Hey @Reus1196 did you find the answer to this query? I'm facing the same trouble as you were/are...

**Reply by PratyushNag** [reply]
Facing the same Error

**Reply by cromewar** [reply]
Try removing the keyword export.

**Comment by nkanyi73** [comment]
I faced the same issues, but I just ignored as despite the underlining, the code still worked well. Its only a problem if you're experiencing errors in other parts of your project. I don't see any error in your .env file.

**Reply by dankorea** [reply]
yes...same problem...keep telling me those variables need declaration...by running scripts, those variables can not be detected either

**Comment by Niferu** [comment, marked as answer]
Hello  @Reus1196 

Have you changed interpreter to `Environment Variable`? If you have extension installed you should be able to pick it by cklicking below in right down corner of your VS Code. Instead of `Environment Variable` it can say `Properties` or `Python` just click on anything you have there.

![image](https://user-images.githubusercontent.com/110537615/191138204-4ee831e7-77d7-4859-ae94-b3dccf84cc5e.png)

Then chose below by writing `.env` in bar:
![image](https://user-images.githubusercontent.com/110537615/191138169-8d6f4b4e-046a-4dd5-986e-33726c923c53.png)
